src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        if (action.type === 'clear') {
+          for (const key in state) {
+            delete state[key];
+          }
+        }
+        break;
+    }
+  }
+
+  return state;
```

Comment:
You are required to change the `state` object, but not to return anything. Delete this return statement.